edx_proctoring/api.py

```diff
@@ -1947,3 +1948,46 @@ def get_student_view(user_id, course_id, content_id,
     if sub_view_func:
         return sub_view_func(exam, context, exam_id, user_id, course_id)
     return None
+
+
+def get_exam_violation_report(course_id, include_practice_exams=False):
+    """
+    Returns proctored exam attempts for the course id, including review details.
+    Violation status messages are aggregated as a list per attempt for each
+    violation type.
+    """
+    attempts_by_code = {
+        attempt['attempt_code']: {
+            'course_id': attempt['proctored_exam']['course_id'],
+            'exam_name': attempt['proctored_exam']['exam_name'],
+            'username': attempt['user']['username'],
+            'email': attempt['user']['email'],
+            'attempt_code': attempt['attempt_code'],
+            'allowed_time_limit_mins': attempt['allowed_time_limit_mins'],
+            'is_sample_attempt': attempt['is_sample_attempt'],
+            'started_at': attempt['started_at'],
+            'completed_at': attempt['completed_at'],
+            'status': attempt['status'],
+            'review_status': None
+        } for attempt in get_all_exam_attempts(course_id)
+    }
+
+    reviews = ProctoredExamSoftwareSecureReview.objects.select_related().filter(
+        exam__course_id=course_id,
+        exam__is_practice_exam=include_practice_exams
+    )
```

Comment:
I *think* we want to specify `proctoredexamsoftwaresecurecomment_set` inside the `select_related()` to avoid fetching more data than we need, but I'm not 100% sure about that. I'll dig a little here.